From d72d3d157bf934029a36829347a0ce3418036209 Mon Sep 17 00:00:00 2001
From: Samuel Dionne-Riel <samuel@dionne-riel.com>
Date: Sat, 19 Dec 2020 00:25:05 -0500
Subject: [PATCH] HACK: disable disp_lcm_suspend

This function crashes the kernel while trying to poweroff the device. In
turn, this means that it cannot be powered off.

It is likely that this is specific to the framebuffer driver being in
use.
---
 .../mediatek/video/mt6785/videox/disp_lcm.c   | 21 ++-----------------
 1 file changed, 2 insertions(+), 19 deletions(-)

diff --git a/drivers/misc/mediatek/video/mt6785/videox/disp_lcm.c b/drivers/misc/mediatek/video/mt6785/videox/disp_lcm.c
index 33c69f3080c2c..05a9b020e5385 100644
--- a/drivers/misc/mediatek/video/mt6785/videox/disp_lcm.c
+++ b/drivers/misc/mediatek/video/mt6785/videox/disp_lcm.c
@@ -1401,25 +1401,8 @@ int disp_lcm_esd_recover(struct disp_lcm_handle *plcm)
 
 int disp_lcm_suspend(struct disp_lcm_handle *plcm)
 {
-	struct LCM_DRIVER *lcm_drv = NULL;
-
-	DISPFUNC();
-	if (_is_lcm_inited(plcm)) {
-		lcm_drv = plcm->drv;
-		if (lcm_drv->suspend) {
-			lcm_drv->suspend();
-		} else {
-			DISP_PR_ERR("FATAL ERROR, lcm_drv->suspend is null\n");
-			return -1;
-		}
-
-		if (lcm_drv->suspend_power)
-			lcm_drv->suspend_power();
-
-		return 0;
-	}
-	DISP_PR_ERR("lcm_drv is null\n");
-	return -1;
+	/* Purposefully neutered; breaks powering off */
+	return 0;
 }
 
 int disp_lcm_resume(struct disp_lcm_handle *plcm)
-- 
2.29.2

